02_knight_and_princess/novikov_dmitry/task_02.cpp

```diff
@@ -0,0 +1,143 @@
+#include <assert.h>
+#include <cassert>
+#include <fstream>
+#include <iostream>
+#include <sstream>
+#include <string>
+#include <unordered_map>
+#include <vector>
+
+constexpr int VERTICES_NUMBER = 14;
+
+using VertexId = int;
+using EdgeId = int;
+
+class Vertex {
+ public:
+  explicit Vertex(const VertexId& new_vertex_id) : id_(new_vertex_id) {}
+
+  void add_edge_id(const EdgeId& new_edge_id) {
+    edge_ids_.push_back(new_edge_id);
+  }
+
+  std::string to_string() const {
+    std::stringstream ss_out;
+    std::string tab_1 = "    ";
+    std::string tab_2 = tab_1 + tab_1;
+    ss_out << tab_2 << "{\n";
+    ss_out << tab_2 << tab_1 << "\"id\": " << id_ << ",\n";
+    ss_out << tab_2 << tab_1 << "\"edge_ids\": [";
+    for (auto it = edge_ids_.begin(); it != edge_ids_.end(); ++it) {
+      if (it != edge_ids_.begin()) {
+        ss_out << ", ";
+      }
+      ss_out << *it;
+    }
+    ss_out << "]\n" << tab_2 << "}";
+    return ss_out.str();
+  }
+
+ private:
+  VertexId id_ = 0;
+  std::vector<EdgeId> edge_ids_;
+};
+
+class Edge {
+ public:
+  Edge(const VertexId& from_vertex_id,
+       const EdgeId& new_edge_id,
+       const VertexId& to_vertex_id)
+      : ver_id1_(from_vertex_id), id_(new_edge_id), ver_id2_(to_vertex_id) {}
+
+  std::string to_string() const {
+    std::stringstream ss_out;
+    std::string tab_1 = "    ";
+    std::string tab_2 = tab_1 + tab_1;
+    ss_out << tab_2 << "{\n";
+    ss_out << tab_2 << tab_1 << "\"id\": " << id_ << ",\n";
+    ss_out << tab_2 << tab_1 << "\"vertex_ids\": [";
+    ss_out << ver_id1_ << ", " << ver_id2_;
+    ss_out << "]\n" << tab_2 << "}";
+    return ss_out.str();
+  }
+
+ private:
+  VertexId ver_id1_;
+  EdgeId id_;
+  VertexId ver_id2_;
+};
+
+class Graph {
+ public:
+  void add_vertex() {
+    vertex_map_.emplace(default_ver_id_, Vertex(default_ver_id_));

```

Comment:
nit: Если используешь `emplace`, достаточно передать аргументы для конструирования объкта. Если же ты сам создаешь объект, то нужно использовать `insert`
```suggestion
    vertex_map_.emplace(default_ver_id_, default_ver_id_);
```